opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -92,24 +93,21 @@ namespace Opm {
             ADB              mob;   // Phase mobility (per cell)
         };
 
-        struct SimulatorData {
+        struct SimulatorData : public Opm::FIPDataEnums {
             SimulatorData(int num_phases);
 
-            enum FipId {
-                FIP_AQUA = Opm::Water,
-                FIP_LIQUID = Opm::Oil,
-                FIP_VAPOUR = Opm::Gas,
-                FIP_DISSOLVED_GAS = 3,
-                FIP_VAPORIZED_OIL = 4,
-                FIP_PV = 5,                    //< Pore volume
-                FIP_WEIGHTED_PRESSURE = 6
-            };
+            using Opm::FIPDataEnums :: FipId ;
```

Comment:
No preference here. I can change it. The inheritance is the normal way I would say.